packages/client/src/objectSet/ObjectSet.ts

```diff
@@ -44,7 +45,15 @@ export interface MinimalObjectSet<Q extends ObjectOrInterfaceDefinition> {
     const A extends Augments,
   >(
     args?: FetchPageArgs<Q, L, R, A>,
-  ) => FetchPageResult<Q, L, R>;
+  ) => Promise<FetchPageResult<Q, L, R>>;
+
+  fetchPageWithErrors: <
+    L extends ObjectOrInterfacePropertyKeysFrom2<Q>,
+    R extends boolean,
+    const A extends Augments,
+  >(
+    args?: FetchPageArgs<Q, L, R, A>,
+  ) => Promise<Result<FetchPageResult<Q, L, R>>>;
```

Comment:
A lot of return types changed just because I know have to wrap with promise since it's not a part of the type